app/models/user.php

```diff
@@ -0,0 +1,67 @@
+<?php
+class User extends AppModel
+{
+    const ERR_DUPLICATE_ENTRY = 1062;
+    public $user_validated = true;
+    public $validation = array(
+        'username' => array(
+            'length' => array(
+                'validate_between', 1, 16,
+            ),
+        ),
+        'password' => array(
+            'length' => array(
+                'validate_between', 1, 16,
+            ),
+        ),
+    );
+
+    public function login(User $user)
+    {
+        if (!$this->isUserExisting() || !$this->isPasswordMatch()) {
+            $this->user_validated = false;
+        }
+        $username = $this->username;
+        $password = $this->password;
+
+        $db = DB::conn();
+        $user = $db->row('SELECT username, password FROM user WHERE username = ? and password = ?', array($username, md5($password)));
+
+        if (!$user) {
+            throw new RecordNotFoundException();
+            $this->user_validated = false;
```

Comment:
Since you are throwing the exception before assigning value to the property, the property's value will not change. Please interchange L31 and L32.